nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_Init.c

```diff
@@ -120,11 +299,34 @@ static NFCSTATUS phNciNfc_ProcessInitRsp(void *pContext, NFCSTATUS Status)
                     pInitRsp->ManufacturerId = pNciContext->RspBuffInfo.pBuff[Offset++];
                     if(pInitRsp->ManufacturerId != 0x00)
                     {
-                        /*Decided by Manufacturer*/
-                        pInitRsp->ManufacturerInfo.Byte0 = pNciContext->RspBuffInfo.pBuff[Offset++];
-                        pInitRsp->ManufacturerInfo.Byte1 = pNciContext->RspBuffInfo.pBuff[Offset++];
-                        pInitRsp->ManufacturerInfo.Byte2 = pNciContext->RspBuffInfo.pBuff[Offset++];
-                        pInitRsp->ManufacturerInfo.Byte3 = pNciContext->RspBuffInfo.pBuff[Offset++];
+                        /* Before allocating a new buffer to hold Manufacturer information data,
+                           check if a valid one is not already available. */
+                        pBuff = pNciContext->InitRspParams.ManufacturerInfo.Buffer;
+                        if (pBuff == NULL)
+                        {
+                            pBuff = (uint8_t *)phOsalNfc_GetMemory(PHNCINFC_CORE_MANUF_INFO_LEN_NCI1x);
```

Comment:
I personally like the goto approach but i believe it would not fit the code style in phNciNfc_Init.c for now.
However, I found multiple places where goto was used.

I believe you spotted that in case of pBuff == NULL after phOsalNfc_GetMemory. I will fix this.